src/Apps/Order/Routes/NewPayment/index.tsx

```diff
@@ -163,7 +166,26 @@ export const NewPaymentRoute: FC<NewPaymentProps & StripeProps> = props => {
             })
             return
           } else {
-            onContinue()
```

Comment:
We used to called `onContinue` here because we wanted to call `fixFailedPayment` a second time, and the logic to handle that response is the same as the first time.

However, on the second call we were not able to fetch the selected credit card ID from state/refs (and I can't figure out why) so I am making that second `fixFailedPayment` call explicitly here.

This means that if the second `fixFailedPayment` call still `requires_action` that we will fail the operation (instead of showing the SCA modal a second time, and I think that's OK.

Reply:
No. If 3DS authentication fails, then the user will not be able to try another credit card. Users effectively get one shot at fixing a failed payment (regardless of whether 3DS is required).

This is a bug that I can reproduce with failure test cards and is not limited to 3DS.